codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -173,6 +177,11 @@ protected ByteBuf allocateBuffer(ChannelHandlerContext ctx, ByteBuf msg, boolean
             bufSize += compressor.maxCompressedLength(curSize) + HEADER_LENGTH;
         }
 
+        if (bufSize > maxEncodeSize) {
+            String errorMsg = "requested encode buffer size (%d bytes) exceeds the maximum allowable size (%d bytes)";
+            throw new EncoderException(String.format(errorMsg, bufSize, maxEncodeSize));
```

Comment:
the style checker kept bitching at me about line length, and this seemed the most readable version; splitting the message across lines looked ugly.